osbs/api.py

```diff
@@ -425,33 +423,70 @@ def _create_build_directly(self, build_request, unique=None):
 
         return BuildResponse(self.os.create_build(build_json).json(), self)
 
+    @retry_on_conflict
+    def _get_or_create_imagestream(self, imagestream_name, build_request):
+        insecure = False
+        source_registry_uri = RegistryURI(build_request.source_registry['url']).docker_uri
+        source_registry_insecure = build_request.source_registry.get('insecure', False)
+
+        docker_image_repo = ImageName.parse(build_request.base_image)
+
+        if not docker_image_repo.registry:
+            docker_image_repo.registry = source_registry_uri
+            insecure = source_registry_insecure
+        else:
+            if docker_image_repo.registry == source_registry_uri:
+                insecure = source_registry_insecure
+
+            else:
+                allowed_registry = False
+
+                if build_request.pull_registries:
+                    for pull_reg in build_request.pull_registries:
+                        if docker_image_repo.registry == pull_reg['url']:
```

Comment:
I'm not sure if there is any agreement on whether the `url` for a pull registry must be just a hostname or if it can be `https://registry.example.com`, but technically, nothing is stopping anyone from using a full URL. It might be safer to check this the same way as the source registry url, i.e. `RegistryURI(...).docker_uri`

Reply:
good point will do

Reply:
fixed